src/app/zap-templates/templates/app/attribute-type.zapt

```diff
@@ -8,4 +8,8 @@ enum {
 {{#zcl_atomics}}
 {{ident}}ZCL_{{asDelimitedMacro name}}_ATTRIBUTE_TYPE = {{asHex atomicId 2}}, // {{description}}
 {{/zcl_atomics}}
+// TODO: workaround for structs, should emit ZCL_STRUCT_ATTRIBUTE_TYPE for structs, this is a issue of zap.
+{{#zcl_structs}}
+{{ident}}ZCL_{{asDelimitedMacro label}}_ATTRIBUTE_TYPE = ZCL_STRUCT_ATTRIBUTE_TYPE,
+{{/zcl_structs}}
```

Comment:
Can you explain what's going on ? I would expect `ZCL_STRUCT_ATTRIBUTE_TYPE` to exist ?

Is it because `ZCL_{{asDelimitedMacro type}}_ATTRIBUTE_TYPE` is using the type name defined into the XML file instead of `STRUCT` ? If so we probably need to resolve the `type` to the base type before passing it to `asDelimitedMacro` instead of adding all those macros.